sql/src/main/java/org/opensearch/sql/sql/parser/AstUnnestBuilder.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.sql.parser;
+
+import lombok.RequiredArgsConstructor;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.opensearch.sql.ast.expression.Alias;
+import org.opensearch.sql.ast.expression.Function;
+import org.opensearch.sql.ast.expression.UnresolvedExpression;
+import org.opensearch.sql.ast.tree.Unnest;
+import org.opensearch.sql.ast.tree.UnresolvedPlan;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.sql.antlr.parser.OpenSearchSQLParserBaseVisitor;
+import org.opensearch.sql.sql.parser.context.QuerySpecification;
+
+@RequiredArgsConstructor
+public class AstUnnestBuilder extends OpenSearchSQLParserBaseVisitor<UnresolvedPlan> {
+
+  /**
+   * Query specification that contains info collected beforehand.
+   */
+  private final QuerySpecification querySpec;
+
+  @Override
+  public UnresolvedPlan visit(ParseTree selectClause) {
+    Unnest unnest = null;
+    for (UnresolvedExpression item : querySpec.getSelectItems()) {
+      // Assume is nested as that's the only function currently
+      // supported as alias in SELECT clause from AstBuilder
+      if (item instanceof Alias) {
```

Comment:
The loop lap skipped if `item` is not an `Alias`. And `visit` returns `null` if all `item`s aren't `Alias`es. Just to confirm that it is corrent.

Reply:
Yes, this may change dependent on if I can achieve pushDown for the ProjectOperator or not though.